packages/@aws-cdk/aws-iotevents/lib/Event.ts

```diff
@@ -0,0 +1,271 @@
+import { Duration, Lazy } from "@aws-cdk/core";
+import { IDetectorModel } from "./DetectorModel";
+import { CfnDetectorModel } from "./iotevents.generated";
+export type AnyEventProperty =
+  | CfnDetectorModel.EventProperty
+  | CfnDetectorModel.TransitionEventProperty;
+
+/**
+ * Event specification
+ *
+ * @export
+ * @interface IEvent
+ */
+export interface IEvent {
+  /**
+   * The event name
+   *
+   * @type {string}
+   * @memberof IEvent
+   */
+  readonly name: string;
+
+  /**
+   * Optional condition for the event
+   *
+   * @type {(string | undefined)}
+   * @memberof IEvent
+   */
+  readonly condition: string | undefined;
+
+  /**
+   * Generate the CFN
+   *
+   * @param {IGrantable} grantable
+   * @returns {AnyEventProperty}
+   * @memberof IEvent
+   * @internal
+   */
+  _cfn(grantable: IGrantable): AnyEventProperty;
+
+  /**
+   * Action to clear a timer
+   *
+   * @param {string} name the timer name
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  clearTimer(name: string): Event;
+
+  /**
+   * Action to publish to Kinesis Firehose
+   *
+   * @param {CfnDeliveryStream} deliveryStream
+   * @param {(string | "\n")} [separator]
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  addFirehose(
+    deliveryStream: CfnDeliveryStream,
+    separator?: string | "\n"
+  ): Event;
+
+  /**
+   * Action to republish into another {@link DetectorModel}
+   *
+   * @param {IDetectorModel} detectorModel
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  republish(detectorModel: IDetectorModel): Event;
+
+  /**
+   * Action to publish to an IoT Topic
+   *
+   * @param {string} topicRule
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  publishToIotTopic(topicRule: string): Event;
+
+  /**
+   * Action to call a lambda function
+   *
+   * @param {IFunction} func
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  addLambda(func: IFunction): Event;
+
+  /**
+   * Action to reset a timer
+   *
+   * @param {string} name
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  resetTimer(name: string): Event;
+
+  /**
+   * Action to set a variable
+   *
+   * @param {string} name
+   * @param {string} value
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  setVariable(name: string, value: string): Event;
+
+  /**
+   * Action to set a timer
+   *
+   * @param {string} name
+   * @param {Duration} duration
+   * @returns {Event}
+   * @memberof IEvent
+   * @throws {Error} if duration is less than 60 seconds
+   */
+  setTimer(name: string, duration: Duration): Event;
+
+  /**
+   * Action to send a SNS
+   *
+   * @param {ITopic} topic
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  addSNS(topic: ITopic): Event;
+
+  /**
+   * Action to publish into an SQS
+   *
+   * @param {IQueue} queue
+   * @param {(boolean | true)} [useBase64]
+   * @returns {Event}
+   * @memberof IEvent
+   */
+  addSQS(queue: IQueue, useBase64?: boolean | true): Event;
+}
+
+/* tslint:disable:callable-types */
```

Comment:
don't think i've seen this rule disabled in the repo before. is there a compelling reason why we need to get around the linter?